tests/unit/test_handlers/test_http_handlers/test_validations.py

```diff
@@ -39,88 +38,77 @@ def test_route_handler_validation_http_method() -> None:
-        )
-
-    with pytest.raises(ImproperlyConfiguredException):
+    with pytest.raises(
+        ImproperlyConfiguredException,
+        match="A status code 204, 304 or in the range below 200 does not support a response body",
+    ):
 
         @delete(path="/")
         def method_with_no_content() -> Dict[str, str]:
             return {}
 
         Litestar(route_handlers=[method_with_no_content])
 
-        method_with_no_content.on_registration(Litestar(), HTTPRoute(path="/", route_handlers=[method_with_no_content]))
-
-    with pytest.raises(ImproperlyConfiguredException):
+    with pytest.raises(
+        ImproperlyConfiguredException,
+        match="A status code 204, 304 or in the range below 200 does not support a response body",
+    ):
 
         @get(path="/", status_code=HTTP_304_NOT_MODIFIED)
         def method_with_not_modified() -> Dict[str, str]:
             return {}
 
         Litestar(route_handlers=[method_with_not_modified])
 
-        method_with_not_modified.on_registration(
-            Litestar(), HTTPRoute(path="/", route_handlers=[method_with_not_modified])
-        )
-
-    with pytest.raises(ImproperlyConfiguredException):
+    with pytest.raises(
+        ImproperlyConfiguredException,
+        match="A status code 204, 304 or in the range below 200 does not support a response body",
+    ):
 
         @get(path="/", status_code=HTTP_100_CONTINUE)
         def method_with_status_lower_than_200() -> Dict[str, str]:
             return {}
 
         Litestar(route_handlers=[method_with_status_lower_than_200])
 
-        method_with_status_lower_than_200.on_registration(
-            Litestar(), HTTPRoute(path="/", route_handlers=[method_with_status_lower_than_200])
-        )
-
     @get(path="/", status_code=HTTP_307_TEMPORARY_REDIRECT)
     def redirect_method() -> Redirect:
         return Redirect("/test")
 
     Litestar(route_handlers=[redirect_method])
 
-    redirect_method.on_registration(Litestar(), HTTPRoute(path="/", route_handlers=[redirect_method]))
-
     @get(path="/")
     def file_method() -> File:
         return File(path=Path("."), filename="test_validations.py")
 
     Litestar(route_handlers=[file_method])
 
-    file_method.on_registration(Litestar(), HTTPRoute(path="/", route_handlers=[file_method]))
-
     assert not file_method.media_type
 
-    with pytest.raises(ImproperlyConfiguredException):
+    with pytest.raises(ImproperlyConfiguredException, match="The 'socket' kwarg is not supported"):
 
         @get(path="/test")
         def test_function_1(socket: WebSocket) -> None:
             return None
 
-        test_function_1.on_registration(Litestar(), HTTPRoute(path="/", route_handlers=[test_function_1]))
+        Litestar([test_function_1])
 
-    with pytest.raises(ImproperlyConfiguredException):
+    with pytest.raises(ImproperlyConfiguredException, match="'data' kwarg is unsupported"):
```

Comment:
nice to have explanation for the end-user, just saying

Reply:
Yup. We should totally overhaul a lot of our error messages to be more specific and explain more.